Source/Details/ASRangeController.mm

```diff
@@ -211,40 +208,13 @@ - (void)_updateVisibleNodeIndexPaths
   
   ASElementMap *map = [_dataSource elementMapForRangeController:self];
 
-  // TODO: Consider if we need to use this codepath, or can rely on something more similar to the data & display ranges
-  // Example: ... = [_layoutController indexPathsForScrolling:scrollDirection rangeType:ASLayoutRangeTypeVisible];
-  NSArray<NSIndexPath *> *visibleNodePaths = [_dataSource visibleNodeIndexPathsForRangeController:self];
```

Comment:
I considered removing this in my diff, but for the historical reasons mentioned, it does add more risk.  I think we need better ASTableView tests before taking on this particular change.

The TODO written here is something I actually did attempt to do, and failed; it had to be reverted.  The comment didn't describe it, but the issue amounted to something like this: not all methods like rectForRowAtIndexPath: actually behave correctly until after a runloop has turned following edit transactions, yet there is some special handling of visibleIndexPaths that makes it more accurate than these other methods.  Then this was combined with some other workarounds as seen in ASTableView.mm